DynamicTablesPkg/Library/Acpi/Common/AcpiSpmiLib/SpmiGenerator.c

```
@@ -0,0 +1,405 @@
+{
+  EFI_STATUS                                 Status;
+  CM_ARCH_COMMON_SPMI_INTERFACE_INFO         *SpmiInfo;
+  CM_ARCH_COMMON_SPMI_INTERRUPT_DEVICE_INFO  *SpmiIntrDeviceInfo;
+  IPMI_GET_DEVICE_ID_RESPONSE                DeviceId;
+
+  ASSERT (This != NULL);
+  ASSERT (AcpiTableInfo != NULL);
+  ASSERT (CfgMgrProtocol != NULL);
+  ASSERT (Table != NULL);
+  ASSERT (AcpiTableInfo->TableGeneratorId == This->GeneratorID);
+  ASSERT (AcpiTableInfo->AcpiTableSignature == This->AcpiTableSignature);
+
+  if ((AcpiTableInfo->AcpiTableRevision < This->MinAcpiTableRevision) ||
+      (AcpiTableInfo->AcpiTableRevision > This->AcpiTableRevision))
+  {
+    DEBUG ((
+      DEBUG_ERROR,
+      "ERROR: SPMI: Requested table revision = %d, is not supported."
+      "Supported table revision: Minimum = %d, Maximum = %d\n",
+      AcpiTableInfo->AcpiTableRevision,
+      This->MinAcpiTableRevision,
+      This->AcpiTableRevision
+      ));
+    return EFI_INVALID_PARAMETER;
+  }
+
+  *Table = NULL;
+
+  Status = GetEArchCommonObjSpmiInterfaceInfo (
+             CfgMgrProtocol,
+             CM_NULL_TOKEN,
+             &SpmiInfo,
+             NULL
+             );
+  if (EFI_ERROR (Status)) {
+    DEBUG ((
+      DEBUG_ERROR,
+      "ERROR: SPMI: Failed to retrieve interface type and base address.\n"
+      ));
+    return Status;
+  }
+
+  AcpiSpmi.InterfaceType = SpmiInfo->InterfaceType;
+  CopyMem (
+    &AcpiSpmi.BaseAddress,
+    &SpmiInfo->BaseAddress,
+    sizeof (EFI_ACPI_6_5_GENERIC_ADDRESS_STRUCTURE)
+    );
+
+  Status = GetEArchCommonObjSpmiInterruptDeviceInfo (
+             CfgMgrProtocol,
+             CM_NULL_TOKEN,
+             &SpmiIntrDeviceInfo,
+             NULL
+             );
+  if (EFI_ERROR (Status)) {
+    DEBUG ((
+      DEBUG_INFO,
+      "INFO: SPMI: The platform does not provide interrupt and PCI device information.\n"
+      ));
+    DEBUG ((
+      DEBUG_INFO,
+      "Using default values (0) for the interrupt and device information.\n"

```

Shouldn't we abort instead of installing an empty table ?

---

this is intentional, platform configuration might not include the legacy EArchCommonObjSpmiInterruptDeviceInfo.
if not provided then the generator will go with the default value which is zero and valid values as per the specification.